### PR TITLE
[backend] Fix playbook filtering cache management (#14285)

### PR DESCRIPTION
**Pull request by MTorbay-Filigran**
### Proposed changes

* Added the applyWithFilters config to the playbook cache filters + added some tests to cover the filtering with every entities such as labels, author, markings...

### Related issues

* #14285 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/container-wrapper-component-test.ts [hunk 1]
```diff
@@ -168,6 +172,21 @@ describe('PLAYBOOK_CONTAINER_WRAPPER_COMPONENT', () => {
       expect(reportResult.object_refs).toEqual([]);
     });
 
+    it('should add only the matching label element with label filter (all-elements)', async () => {
+      const result = await PLAYBOOK_CONTAINER_WRAPPER_COMPONENT.executor(testExecutor({
+        mainId: MALWARE_ID,
+        bundleObjects: BUNDLE_OBJECTS(),
+        configuration: componentConfig({
+          applyToElements: playbookBundleElementsToApply.allElements.value,
+          applyWithFilters: filterLabel,
```
Comment (Archidoit) [inline opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/container-wrapper-component-test.ts:181]: add a test with applyWithFilters = undefined or null

Comment (MTorbay-Filigran) [reply opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/container-wrapper-component-test.ts:181]: Normally we covered the case when there is a bundle with multiple elements, scoped with applyToElements and no filters such as here : 
```
it('should wrap all elements in Container except main element', async () => {
    const result = await PLAYBOOK_CONTAINER_WRAPPER_COMPONENT.executor(testExecutor({
      mainId: MALWARE_ID,
      bundleObjects: BUNDLE_OBJECTS(),
      configuration: componentConfig({
        applyToElements: playbookBundleElementsToApply.allExceptMain.value,
      }),
    }));

    const reportResult = result.bundle.objects
      .find((element) => element.type === 'report') as StixContainer;
    expect(reportResult.object_refs).toEqual([CAMPAIGN_ID, REL_ID]);
  });
```

File: opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/create-indicator-component-test.ts [hunk 2]
```diff
@@ -13,14 +13,16 @@ import type { StixInternalExternalReference } from '../../../../src/types/stix-2
 const OBSERVABLE_ID = 'domain-name--a1b2c3d4-0000-0000-0000-000000000001' as const;
 const INTRUSION_SET_ID = 'intrusion-set--1ad04810-ab05-5873-96f5-a89d19607e1c' as const;
 const REPORT_ID = 'report--b4754e7d-88b4-51d9-aac4-86edaad66c4d' as const;
+const LABEL_ID = '98d475ca-0f72-4878-8d64-de2e094f007e';
 
 const FAKE_PATTERN = "[domain-name:value = 'malicious.example.com']";
 
-const domainObservable = (id: StixId = OBSERVABLE_ID) =>
-  testBundleObject<StixCyberObject & { value: string }>({
+const domainObservable = (id: StixId = OBSERVABLE_ID, withLabel?: boolean) =>
```
Comment (lndrtrbn) [inline opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/create-indicator-component-test.ts]: It would be a better pattern to use an approach similar to the function `componentConfig` _(see L182 in manipulate test file)_ we use to construct the configuration with a partial object that you can directly spread into the base object instead of making a ternary for each attribute you want to add

File: opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/create-indicator-component-test.ts [hunk 3]
```diff
@@ -13,14 +13,16 @@ import type { StixInternalExternalReference } from '../../../../src/types/stix-2
...
+  testBundleObject<StixCyberObject & { value: string; labels: string[] }>({
     id,
     type: 'domain-name',
     value: 'malicious.example.com',
+    ...(withLabel ? { labels: [LABEL_ID] } : {}),
```
Comment (lndrtrbn) [inline opencti-platform/opencti-graphql/tests/03-integration/01-database/playbookComponents/create-indicator-component-test.ts]: I am not sure of putting labels at this place. In STIX objects they are normally in the extensions. Do you find something that said otherwise?

**Review by lndrtrbn — APPROVED** [review]
